Migrate to Vite

swouf opened the pull request:
Migrate to Vite for building the app. Upgrades some dependencies on the way.

Close #141

swouf committed:
chore(deps): upgrade all deps

swouf committed:
build: migrate to vite

adapt config and tests

swouf committed:
chore(deps): add resolution for immutable and minor upgrade

swouf committed:
ci: refactor workflows for vite

swouf committed:
chore(context): add type annotations

swouf committed:
ci(test): fix cypress workflow

spaenleh commented on .github/workflows/deploy-prod.yml:
```diff
@@ -1,14 +1,13 @@
-name: Deploy to production environment
+name: Deploy to staging environment
```
You have a typo here, it should be `production`
```suggestion
name: Deploy to production environment
```

spaenleh commented on .github/workflows/deploy-stage.yml:
```diff
@@ -6,10 +6,9 @@ on:
-  deploy-app-to-staging:
-    name: Deploy to staging
+  deploy-app:
+    name: Deploy to dev
```
```suggestion
    name: Deploy to staging
```

spaenleh commented on package.json:
```diff
@@ -26,117 +26,114 @@
+    "dev:mock": "VITE_MOCK_API=true && yarn vite",
+    "dev": "yarn vite",
+    "start:test": "yarn vite --mode test",
+    "build": "NODE_OPTIONS=--max_old_space_size=16384 && yarn vite build",
```
You should not need that option. Try removing it, and checking your versions of sdk, immutable etc to see if they match.

spaenleh commented on package.json:
```diff
@@ -26,117 +26,114 @@
-    "build:dev": "env-cmd -f ./.env.dev react-scripts --max_old_space_size=4096 build",
-    "test:cra": "react-scripts test",
-    "eject": "react-scripts eject",
+    "dev:mock": "VITE_MOCK_API=true && yarn vite",
+    "dev": "yarn vite",
+    "start:test": "yarn vite --mode test",
+    "build": "NODE_OPTIONS=--max_old_space_size=16384 && yarn vite build",
+    "preview": "yarn vite preview",
```
I think you can omit the `yarn` for these commands, `vite build` should be enough for example.

spaenleh commented on package.json:
```diff
@@ -26,117 +26,114 @@
+    "@typescript-eslint/eslint-plugin": "5.62.0",
+    "@typescript-eslint/parser": "5.62.0",
+    "@vitejs/plugin-react": "4.0.3",
     "codacy-coverage": "3.4.0",
```
I don't think we use this anymore.

spaenleh commented on package.json:
```diff
@@ -26,117 +26,114 @@
     "husky": "~8.0.3",
-    "interpolate-html-plugin": "4.0.0",
-    "json-server": "0.17.3",
     "npm-run-all": "4.1.5",
```
I think this could be removed also.

spaenleh reviewed (changes requested):
Very nice ! ⭐ Just a few comments, and some typos here and there.

swouf commented on .github/workflows/deploy-dev.yml:
```diff
@@ -0,0 +1,48 @@
+  deploy-app:
+    name: Deploy to dev
+    runs-on: ubuntu-latest
+    environment: development
```
Okay, I didn't know about that actually. I'll keep it then and will add the secrets to the development env.

swouf committed:
chore(ci): fix typos in workflows

Co-authored-by: Basile Spaenlehauer <spaenleh@gmail.com>

swouf committed:
chore: apply review comments

https://github.com/graasp/graasp-app-sticky-notes/pull/142#pullrequestreview-1562934602
The commit was amended.

spaenleh commented:
For your tests, you might want to lower the typing speed, it looks like some keystrokes are not sent to the text editor.

spaenleh reviewed (approved):
Looks good, let me know if the tests pass ✅
